src/test/java/uk/gov/justice/hmpps/casenotes/services/CaseNoteTypeMergerTest.java

```diff
@@ -49,7 +49,45 @@ public void testSimpleMerge() {
+
+        final var list1 = List.of(
+                CaseNoteType.builder().code("OBS").description("Observation").activeFlag("Y").subCodes(
+                        List.of(CaseNoteType.builder().code("GEN").description("General").activeFlag("Y").build(), CaseNoteType.builder().code("SP").description("Special").activeFlag("N").build())
+                ).build(),
+                CaseNoteType.builder().code("DRR").description("Drug Rehabilitation Requirement").activeFlag("N").subCodes(
+                        List.of(CaseNoteType.builder().code("DCOUN").description("Drug Counselling Session").activeFlag("Y").build(), CaseNoteType.builder().code("DTEST").description("Drug Test").activeFlag("N").build())
+                ).build()
+        );
 
+        final var list2 = List.of(
+                CaseNoteType.builder().code("OBS").description("Observation").activeFlag("Y").subCodes(
+                        List.of(CaseNoteType.builder().code("NEW").description("New Stuff").activeFlag("Y").build(), CaseNoteType.builder().code("GEN").activeFlag("Y").description("Different Gen").activeFlag("Y").build())
+                ).build(),
+                CaseNoteType.builder().code("POM").description("POM Stuff").activeFlag("Y").subCodes(
+                        List.of(CaseNoteType.builder().code("SPC").description("Special").activeFlag("Y").build(), CaseNoteType.builder().code("GEN").description("General").activeFlag("Y").build())
+                ).build()
+        );
 
+        List<CaseNoteType> resultantList = merger.mergeAndSortList(list1, list2);
+
+        assertThat(resultantList).hasSize(3);
+
+        assertThat(resultantList).containsExactly(
+                List.of(
+                        CaseNoteType.builder().code("DRR").description("Drug Rehabilitation Requirement").activeFlag("N").subCodes(
+                                List.of(CaseNoteType.builder().code("DCOUN").description("Drug Counselling Session").activeFlag("N").build(), CaseNoteType.builder().code("DTEST").description("Drug Test").activeFlag("N").build())
+                        ).build(),
+                        CaseNoteType.builder().code("OBS").description("Observation").activeFlag("Y").subCodes(
+                                List.of(CaseNoteType.builder().code("GEN").description("Different Gen").activeFlag("Y").build(), CaseNoteType.builder().code("NEW").description("New Stuff").activeFlag("Y").build(), CaseNoteType.builder().code("SP").description("Special").activeFlag("N").build())
+                        ).build(),
+                        CaseNoteType.builder().code("POM").description("POM Stuff").activeFlag("Y").subCodes(
+                                List.of(CaseNoteType.builder().code("GEN").description("General").activeFlag("Y").build(), CaseNoteType.builder().code("SPC").description("Special").activeFlag("Y").build())
+                        ).build()
+                ).toArray(new CaseNoteType[0])
```

Comment:
not sure why converting here into an array?

Reply:
if you don't convert into a list in the first place you won't then need to `toArray` here